java/src/main/java/org/rocksdb/CompactRangeOptions.java

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+package org.rocksdb;
+
+/**
+ * CompactRangeOptions is used by CompactRange() call. In the documentation of the methods "the compaction" refers to
+ * any compaction that is using this CompactRangeOptions.
+ */
+public class CompactRangeOptions extends RocksObject {
+
+  // For level based compaction, we can configure if we want to skip/force bottommost level compaction.
+  // The order of this neum MUST follow the C++ layer. See BottommostLevelCompaction in db/options.h
+  public enum BottommostLevelCompaction {
+    /**
+     * Skip bottommost level compaction
+     */
+    kSkip,
+    /**
+     * Only compact bottommost level if there is a compaction filter. This is the default option
+     */
+    kIfHaveCompactionFilter,
+    /**
+     * Always compact bottommost level
+     */
+    kForce,;
+
+    /**
+     * Returns the native C++ rocks enum value
+     * @return the native C++ rocks enum value
+     */
+    public int rocksId() {
+      return this.ordinal(); // identical with the C++ layer
```

Comment:
I would like to avoid using the ordinal as it is fragile in terms of changing if options are added/removed in future. Please could you use an explicit value for each enum value and update `portal.h` appropriately?

See `AccessHint.java` as one such example of how that should work